packages/clarity-js/src/data/extract.ts

```diff
@@ -86,19 +88,25 @@ export function compute(): void {
                 let selectorData = selectors[key];
                 for (let s in selectorData) {
                     let selectorKey = parseInt(s);
-                    let nodes = document.querySelectorAll(selectorData[selectorKey]) as NodeListOf<HTMLElement>;
+                    let selector = selectorData[selectorKey];
+                    if (selector.startsWith(Constant.At)){
+                        shouldMask = true;
+                        selector = selector.slice(1, selector.length);
+                    }
+                    let nodes = document.querySelectorAll(selector) as NodeListOf<HTMLElement>;
                     if (nodes) {
                         let text = Array.from(nodes).map(e => e.textContent)
-                        update(key, selectorKey, text.join(Constant.Seperator).substring(0, Setting.ExtractLimit));
+                        let content = shouldMask ? hash(text.join(Constant.Seperator)).trim() : text.join(Constant.Seperator);
+                        update(key, selectorKey, content.slice(0, Setting.ExtractLimit));
```

Comment:
nit: dothe .join(Constant.seperator).trim() on line 100 so you dont have to have it twice